src/muya/lib/contentState/searchCtrl.js

```diff
@@ -34,11 +34,23 @@ const matchString = (text, value, options) => {
 }
 
 const searchCtrl = ContentState => {
+  ContentState.prototype.buildRegexValue = function (match, value) {
+    const groups = value.match(/\$(\d)*/g)
+    if (groups) {
+      for (const regexGroup of groups) {
+        const groupIndex = regexGroup[1] - 1
```

Comment:
Here are two issues: the first issue occurs if you match `$-1` or `$a` and the second is just style. Please check whether `regexGroup[1]` is defined inside the loop (`if (regexGroup[1]) { loop-body }`) and parse the integer like `Number(regexGroup[1]) - 1`.

Reply:
>Here are two issues: the first issue occurs if you match $-1 or $a and the second is just style. Please check whether regexGroup[1] is defined inside the loop (if (regexGroup[1]) { loop-body }) and parse the integer like Number(regexGroup[1]) - 1.

@fxha I don't quite understand your first bad case, can you give me an example? There is a second one I also hope to give an example.

@ihipi are you still have time on this PR?

Reply:
- `\d` allows to match negative numbers. Ok, just `$-` because we are restricted to a single digit only.
- `$0` should return the whole match but returns `undefined`.
  - Example: Text: `abc`; Search: `(a)b(c)`: Replacement: `x$0y` -> expected `xabcy`
- Unable to escape `$` inside the replacement; use look-behind to ignore escaped character (`\$`)
- Replacement Regex:
  - `*` should be `+` or `{1}`
  - `/\$(\d)*/g` should be `/\$(\d+)/g` to allow digits >=10?
  - `$a` is currently a valid match, which result in `regexGroup[1]` to be undefined.

@Jocs We could also merge this if you fix the remaining issues soon.

Reply:
Cases larger than 10 should rarely appear? So we don't need to handle it.